Add function for μ stability factor

asandwhich opened the pull request:
This adds a function to calculate the standalone μ
stability factor as defined in [1].

[1]: M. L. Edwards and J. H. Sinsky, "A new criterion for linear 2-port
     stability using a single geometrically derived parameter," in IEEE
     Transactions on Microwave Theory and Techniques, vol. 40, no. 12,
     pp. 2303-2311, Dec. 1992. doi: 10.1109/22.179894

asandwhich committed:
Add function for μ stability factor

This adds a function to calculate the standalone μ
stability factor as defined in [1].

[1]: M. L. Edwards and J. H. Sinsky, "A new criterion for linear 2-port
     stability using a single geometrically derived parameter," in IEEE
     Transactions on Microwave Theory and Techniques, vol. 40, no. 12,
     pp. 2303-2311, Dec. 1992. doi: 10.1109/22.179894

asandwhich commented on src/Marconi.jl:
```diff
@@ -339,7 +339,7 @@ end
     testMagDelta(network)
 
-Returns a vector of `Δ`, the determinant of the scattering matrix.
+Returns a vector of `|Δ|`, the magnitude of the determinant of the scattering matrix.
```
This can be separated into a different commit if atomicity is strongly desired.

kiranshila commented:
Good stuff! Do you mind updating the doc page for it under stability?

Thank you for the commit!

asandwhich committed:
Add documentation for μ stability condition

This adds an entry beneath the Rollet stability condition
to highlight that the μ stability condition can be calculated
in a similar way.

kiranshila commented:
Awesome! Thanks a ton!

By the way, there will probably be some structural changes soon to a lot of stuff - stick around if you'd like to help, I really appreciate it!

asandwhich commented:
Sure, will see if I can do anything. 

One suggestion for some structural changes, apologies if this has already been considered, would be to store the network parameters in a 3 dimensional matrix, as opposed to an array of the 2-d arrays. Would allow for indexing behavior similar to matlab: 
some_network.s_params(1, 1, :)
Which would allow for easy use of the auto-vectorized functions such as 
abs.(some_network.s_params(1,1,:)).^2

If it sounds desirable I can maybe get a pull request up by next week.

kiranshila commented:
Yeah I think I am certainly going to rework the network, especially touchstone parsing. I found this [library](https://github.com/PainterQubits/Touchstone.jl) which we may want to incorporate.
